### PR TITLE
Remove quoted name splices

### PR DESCRIPTION
**Pull request by nicolasstucki**
The current encoding of name splices is quite ad-hoc and intrusive. It changes the meaning
of a name that starts with `$` which does not align with other kinds of splices. It also does
not align with the names that may be shown by `Expr.show`.

<del>Based on #8692.</del>

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Additionally, this feature does not provide any semantic meaning to the names hence it does not align with the quotes and splices abstractions.

It is always possible to recover this name using reflection.

File: library/src/scala/internal/quoted/Matcher.scala [hunk 1]
```diff
@@ -247,28 +239,21 @@ private[quoted] object Matcher {
             tycon1 =?= tycon2 && args1 =?= args2
 
           case (ValDef(_, tpt1, rhs1), ValDef(_, tpt2, rhs2)) if checkValFlags() =>
-            val bindMatch =
-              if (hasBindAnnotation(pattern.symbol) || hasBindTypeAnnotation(tpt2)) bindingMatch(scrutinee.symbol)
-              else matched
             def rhsEnv = summon[Env] + (scrutinee.symbol -> pattern.symbol)
-            bindMatch && tpt1 =?= tpt2 && treeOptMatches(rhs1, rhs2)(using summon[Context], rhsEnv)
+            tpt1 =?= tpt2 && treeOptMatches(rhs1, rhs2)(using summon[Context], rhsEnv)
```
Comment (liufengyun) [inline library/src/scala/internal/quoted/Matcher.scala:243]: The pattern matching on definitions up to alpha-conversion 👍

**Comment by nicolasstucki** [comment]
We need to rebootstrap the compiler to have #8692 in the reference compiler.

**Comment by smarter** [comment]
> We need to rebootstrap the compiler to have #8692 in the reference compiler.

Alternatively, you could leave the definitions needed by the non-bootstrapped compiler in the library for now, or move them to `library/src-non-bootstrapped/`

**Comment by nicolasstucki** [comment]
I kept the annotation for now as it is internal and will not change anything. I left a `TODO` to remove it when we rebootstrap.

**Pull request merged**